src/main/java/org/folio/fql/model/field/FqlField.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.fql.model.field;
+
+import org.folio.fql.deserializer.FqlParsingException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Value;
+
+@Value
+public class FqlField {
+
+  private final String columnName;
+  private final List<SubField> subFields;
+
+  public FqlField(String field) {
+    int firstSubFieldIndex = getIndexOfSubfield(field);
+
+    this.subFields = new ArrayList<>();
+
+    if (firstSubFieldIndex == -1) {
+      this.columnName = field;
+    } else {
+      this.columnName = field.substring(0, firstSubFieldIndex);
+      parseSubFields(field, field.substring(firstSubFieldIndex));
+    }
+  }
+
+  private static int getIndexOfSubfield(String str) {
```

Comment:
Since there's special meaning around `-1` getting returned here, a javadoc on this method would be super helpful for making this class easier to read